File: gene-tree/src/app/modules/relative/models/request-create-update.model.ts
import { PersonDetailsModel } from '../../person/models/person-details.model';
import { Guid } from 'guid-typescript';
import { GenericPersonModel } from '../../person/models/generic-person.model';

export class RequestCreateUpdateModel {
  id?: number;
  senderUserId: Guid;
  receiverUser: GenericPersonModel;
  receiverReferenceInSenderTreeId: number;
}

File: genealogy-tree-API/GenealogyTree.Domain/DTO/Request/RequestCreateUpdateModel.cs
using GenealogyTree.Domain.DTO.Person;
using System;

namespace GenealogyTree.Domain.DTO.Request
{
    public class RequestCreateUpdateModel
    {
        public int? Id { get; set; }
        public Guid SenderUserId { get; set; }
        public GenericPersonModel ReceiverUser { get; set; }
        public int ReceiverReferenceInSenderTreeId { get; set; }
    }
}
